design-documents/add-redeem-method.md

```diff
@@ -0,0 +1,13 @@
+## Add GiftCardRedeemInterface interface as a service contract
+#### Why?
+Redeeming a gift card is an existing part of the gift card module functionality,
+API should reflect that and allow for client code to execute it and/or use it as Web API.
+#### Situation
+Right now "redeem" method only exists in the Giftcardaccount model and
+actually does not contain all the logic required to redeem a gift card - only Customer\Index constroller has all the logic -
+it would be much better if a service had a method for this functionality with all the required logic in it
+and to remove the logic from the controller by leaving only the GiftCardRedeemInterface::redeem() call.
```

Comment:
Name should be noun. How about `GiftCardRedeemInterface`? I would prefer `RedeemInterface` because GiftCard is in the namespace and it's clear but might be better to keep it because other interfaces in this directory have it.

Reply:
@melnikovi I did name it GiftCardRedeemInterface, did you mean to write something else?

Reply:
I'm sorry, I meant to write `GiftCardRedeemableInterface` or `RedeemableInterface`.

Reply:
@melnikovi Maybe RedeemManagerInterface? RedeemableInterface sounds like an entity that can be redeemed

Reply:
Let's call it GiftCardRedeemerInterface.

Reply:
ok